biobalm/_sd_attractors/attractor_candidates.py

```diff
@@ -0,0 +1,760 @@
+"""
+Provides functions that are used to identify and reduce the set of attractor candidate states.
+
+Note that most of these functions assume they are operating on the network or petri net
+percolated to the subspace of the relevant node.
+"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal, cast
+
+if TYPE_CHECKING:
+    from biobalm.succession_diagram import SuccessionDiagram
+    from biobalm.types import BooleanSpace
+    from networkx import DiGraph  # type: ignore
+
+import random
+import biobalm
+from biodivine_aeon import Bdd, AsynchronousGraph
+from biobalm.trappist_core import compute_fixed_point_reduced_STG
+from biobalm.symbolic_utils import state_list_to_bdd, valuation_to_state, state_to_bdd
+
+try:
+    PINT_AVAILABLE = True
+    import biobalm._pint_reachability as pint
+except ModuleNotFoundError:
+    PINT_AVAILABLE = False
+
+
+CANDIDATES_HARD_LIMIT = 100_000
```

Comment:
Do we really need these global parameters? The problem is that if a user wants to change them, their linter will complain because we've indicated that they are global constants (by writing in all caps... because having a `const` keyword wouldn't be very "Pythonic" 🙄).